### ISSUE TITLE
Application freezes when sharing local drive but entering machine credentials wrong

### ISSUE THREAD

**Issue by OscarVanL** [docker/for-win#4471]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: BFA4B8CD-3A32-4C6B-8204-6FEB8AC88C0F/20190808093519

### Expected behaviour

Popup stating I've entered my machine's credentials incorrectly, so it could not add C as a Shared Drive.

### Actual behaviour

The program initially goes unresponsive (greyed out) and Windows asks if I'd like to kill it or wait. If I wait the program eventually stops being greyed out but none of the buttons do anything, even the X close button. I have to kill the application in task manager and restart it.

### Information

  - Windows Version: Windows 10 Education (identical to Enterprise), 1903, 16/07/2019, 18362.239
  - Docker Desktop Version: 2.1.0.0 (36874)

### Steps to reproduce the behavior

  1. Open Docker Desktop Settings
  2. Click Shared Drives
  3. Add a drive (C drive in my case).
  4. Click Apply
  5. Enter the incorrect password and click OK.

**Comment by mikeparker** [comment]
Unfortunately I can't see any errors in your diagnostic, and I can't reproduce, I get a new popup with the error message:

![image](https://user-images.githubusercontent.com/1557887/62706245-0987f600-b9e7-11e9-989c-f69ee8826c6b.png)

I'll keep looking..

**Comment by OscarVanL** [comment]
I just had this update for 2.1.0.1 popup a few minutes ago and now it's fixed 
![image](https://user-images.githubusercontent.com/7999692/62721409-094a2380-ba04-11e9-8895-0f1ca5fcd347.png)

According to the [release notes](https://docs.docker.com/docker-for-windows/release-notes/), that was just released today/yesterday (considering timezones I'm not sure if that was after I submitted this bug or just before?).

I created this bug report immediately after fresh-installing Docker from the link on your website, so maybe another contributor was able to recreate the bug and fixed it?

After applying the update I get the same popup as you informing me that I entered an Invalid login/password, so this is fixed! (The fastest fix for a bug on a project this big I've ever seen, remarkable really :P)
